frontend/src/pages/projects/screens/spawner/SpawnerPage.tsx

```diff
@@ -68,19 +72,30 @@ const SpawnerPage: React.FC<SpawnerPageProps> = ({ existingNotebook }) => {
   const [supportedAcceleratorProfiles, setSupportedAcceleratorProfiles] = React.useState<
     string[] | undefined
   >();
-  const [storageDataWithoutDefault, setStorageData] = useStorageDataObject(existingNotebook);
-
   const [defaultStorageClass] = useDefaultStorageClass();
   const preferredStorageClass = usePreferredStorageClass();
   const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
   const defaultStorageClassName = isStorageClassesAvailable
     ? defaultStorageClass?.metadata.name
     : preferredStorageClass?.metadata.name;
-  const storageData = useMergeDefaultPVCName(
-    storageDataWithoutDefault,
-    k8sNameDescriptionData.data.name,
-    defaultStorageClassName,
-  );
+  const defaultNotebookSize = useDefaultPvcSize();
+  const [storageData, setStorageData] = React.useState<StorageData[]>([
+    {
+      storageType: existingNotebook ? StorageType.EXISTING_PVC : StorageType.NEW_PVC,
+      creating: {
+        nameDesc: {
+          name: k8sNameDescriptionData.data.name || defaultClusterStorage.name,
+          description: defaultClusterStorage.description,
+        },
+        size: defaultClusterStorage.size || defaultNotebookSize,
+        storageClassName: defaultStorageClassName || defaultClusterStorage.storageClassName,
+        mountPath: defaultClusterStorage.mountPath,
+      },
+      existing: {
+        storage: getRootVolumeName(existingNotebook),
+      },
+    },
```

Comment:
so this exisitng data should include all attached PVCs. create some hook that is basically the inverse of `useRelatedNotebooks` called `useRelatedPVCs`

Reply:
this may also have to be restructured then because existing would need to be able to update the mount path

my suggestions is something along the lines of

```tsx
{
  storageType: existingNotebook ? StorageType.EXISTING_PVC : StorageType.NEW_PVC,
  name: getRootVolumeName(existingNotebook),
  nameDesc: {
     name: existingName || k8sNameDescriptionData.data.name || defaultClusterStorage.name,
     description: existingDescription || defaultClusterStorage.description,
  },
  size: existingSize || defaultClusterStorage.size || defaultNotebookSize,
  storageClassName: existingStorageClassName || defaultStorageClassName || defaultClusterStorage.storageClassName,
  mountPath: existingMountPath || defaultClusterStorage.mountPath,
}
```

Reply:
Updated, but using a variation of what was suggested here.